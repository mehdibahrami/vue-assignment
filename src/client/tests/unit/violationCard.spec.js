import { shallowMount } from "@vue/test-utils";
import ViolationCard from "@/components/cards/ViolationCard.vue";

describe("ViolationCard", () => {
  let wrapper;

  it("renders correctly", () => {
    // shallow mounting the main component
    // Stub card component replacing with its template to load slotted component
    wrapper = shallowMount(ViolationCard, {
      global: {
        stubs: {
          Card: {
            template: '<div class="card-body"><slot></slot></div>',
          },
        },
      },
      props: {
        badge: "11/23/2017, 15:14:17",
        leftTitle: "Speed",
        leftValue: 52,
        mainTitle: "Catharijnesingel",
        rightTitle: "Allowed Speed",
        rightValue: 45,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
