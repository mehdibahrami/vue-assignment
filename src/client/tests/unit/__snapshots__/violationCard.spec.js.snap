// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViolationCard renders correctly 1`] = `
<div
  badge="11/23/2017, 15:14:17"
  class="card-body"
  haspadding="true"
  lefttitle="Speed"
  leftvalue="52"
  maintitle="Catharijnesingel"
  righttitle="Allowed Speed"
  rightvalue="45"
>
  
  <!-- Violations -->
  
  <!-- Empty -->
  <div
    class="empty-box"
  >
    No Violation!
  </div>
  
  
</div>
`;
